### PR TITLE
feat: decoding of url-safe base64 encoding for oobnotes

### PR DESCRIPTION
**Pull request by dpc**
To help the transition, we first implement decoding of url-safe base64 version (accepting both that and old version) and after a while we'll switch the encoding to url-safe one.

Re #4040

### CODE REVIEW

File: modules/fedimint-mint-client/src/lib.rs [hunk 1]
```diff
@@ -227,12 +228,21 @@ impl Decodable for OOBNotes {
...
     type Err = anyhow::Error;
 
     /// Decode a set of out-of-band e-cash notes from a base64 string.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let bytes = base64::decode(s)?;
+        let bytes = if let Ok(bytes) = BASE64_URL_SAFE.decode(s) {
+            bytes
+        } else {
+            base64::engine::general_purpose::STANDARD.decode(s)?
+        };
```
Comment (elsirion) [inline modules/fedimint-mint-client/src/lib.rs:245]: Looked at the standard and there seems to be no case where there is a valid non-URL-safe base64 string that is also a valid url-safe base64 string encoding different data.

**Review by elsirion — APPROVED** [review]
Great idea!

**Comment by elsirion** [comment]
Let's backport so we can make the window between introducing decoding and switching encoding longer/allow earlier switching without breaking backwards compat guarantees.

**Comment by elsirion** [comment]
I see you were already a step ahead :laughing: #4260

**Review by maan2003 — APPROVED** [review]

**Pull request merged**